### ISSUE TITLE
no-unused-vars fails on default parameter with matching name

### ISSUE THREAD

**Issue by jlsjonas** [eslint/eslint#12354]
**Tell us about your environment**

* **ESLint Version:** 6.4.0
* **Node Version:** 10.16.3
* **npm Version:** n.a.

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
airbnb
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const rp = require('request-promise-native');

class Foo {
  constructor(foo, bar, rp = rp) {
    this.bar = rp;
  }
}
```

```bash
ide-integrated
```

**What did you expect to happen?**
no error

**What actually happened? Please include the actual, raw output from ESLint.**
esLint doesn't register that `rp` is being used (as default for the equally named parameter `rp`) in the constructor

**Are you willing to submit a pull request to fix this bug?**
If I'd know how to, yes.

**Comment by mdjermanovic** [comment]
Hi @jlsjonas, thanks for the issue!

That default assignment would produce a `ReferenceError`, since the `rp` is already shadowed.

**Comment by jlsjonas** [comment]
🤔 Indeed, didn't test the default path; sorry 😁
